compiler/rustc_crate/src/lib.rs

```diff
@@ -0,0 +1,46 @@
+#![feature(int_error_matching)]
+#![feature(min_specialization)]
+#![feature(once_cell)]
+#![feature(option_expect_none)]
+#![feature(or_patterns)]
+
+#[macro_use]
+extern crate bitflags;
+#[macro_use]
+extern crate rustc_macros;
+
+use rustc_data_structures::fx::{FxHashMap, FxHashSet};
+use rustc_span::symbol::Symbol;
+
+pub mod codegen_fn_attrs;
+pub mod cstore;
+pub mod dependency_format;
+pub mod ich;
+pub mod limits;
+pub mod privacy;
+pub mod stability;
```

Comment:
This seems like a very disjoint set of modules -- are they all interdependent? If not, it seems odd to move them into a new crate as a single group.

Reply:
I started by aiming at cstore, which is needed for ICH, and with what I could in rustc_middle::middle.
I welcome suggestions on the correct place for each module.